staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editoptions_test.go

```diff
@@ -277,3 +281,241 @@ func TestManagedFieldsExtractAndRestore(t *testing.T) {
+								Object: map[string]interface{}{
+									"apiVersion": "v1",
+									"kind":       "Thingy",
+									"metadata": map[string]interface{}{
+										"uid":       "12345",
+										"namespace": "ns",
+										"name":      "after",
+									},
+									"spec": map[string]interface{}{},
+								},
+							},
+							Mapping: unregMapping,
+						},
+					},
+				},
+				results: &editResults{},
+			},
+			checkErr: expectedErr,
+		},
+		{
+			name:   "kind-diff",
+			fields: fields{},
+			args: args{
+				originalInfos: []*resource.Info{
+					{
+						Namespace: "ns",
+						Name:      "myname",
+						Object: &unstructured.Unstructured{
+							Object: map[string]interface{}{
+								"apiVersion": "v1",
+								"kind":       "Thingy",
+								"metadata": map[string]interface{}{
+									"uid":       "12345",
+									"namespace": "ns",
+									"name":      "myname",
+								},
+								"spec": map[string]interface{}{},
+							},
+						},
+						Mapping: unregMapping,
+					},
+				},
+				patchVisitor: &testVisitor{
+					updatedInfos: []*resource.Info{
+						{
+							Namespace: "ns",
+							Name:      "myname",
+							Object: &unstructured.Unstructured{
+								Object: map[string]interface{}{
+									"apiVersion": "v1",
+									"kind":       "OtherThingy",
+									"metadata": map[string]interface{}{
+										"uid":       "12345",
+										"namespace": "ns",
+										"name":      "myname",
+									},
+									"spec": map[string]interface{}{},
+								},
+							},
+							Mapping: unregMapping,
+						},
+					},
+				},
+				results: &editResults{},
+			},
+			checkErr: expectedErr,
+		},
+		{
+			name:   "apiver-diff",
+			fields: fields{},
+			args: args{
+				originalInfos: []*resource.Info{
+					{
+						Namespace: "ns",
+						Name:      "myname",
+						Object: &unstructured.Unstructured{
+							Object: map[string]interface{}{
+								"apiVersion": "v1",
+								"kind":       "Thingy",
+								"metadata": map[string]interface{}{
+									"uid":       "12345",
+									"namespace": "ns",
+									"name":      "myname",
+								},
+								"spec": map[string]interface{}{},
+							},
+						},
+						Mapping: unregMapping,
+					},
+				},
+				patchVisitor: &testVisitor{
+					updatedInfos: []*resource.Info{
+						{
+							Namespace: "ns",
+							Name:      "myname",
+							Object: &unstructured.Unstructured{
+								Object: map[string]interface{}{
+									"apiVersion": "v1alpha1",
+									"kind":       "Thingy",
+									"metadata": map[string]interface{}{
+										"uid":       "12345",
+										"namespace": "ns",
+										"name":      "myname",
+									},
+									"spec": map[string]interface{}{},
+								},
+							},
+							Mapping: unregMapping,
+						},
+					},
+				},
+				results: &editResults{},
+			},
+			checkErr: expectedErr,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			o := &EditOptions{
```

Comment:
```suggestion
			o := &EditOptions{}
```

Reply:
@eddiezane Thanks, I've made the suggested changes.